irondb-relay.md

```diff
@@ -0,0 +1,609 @@
+    
+    #### A note on flushing results to IRONdb
+    
+    >The very first row that matches and creates an aggregation "window" will
+    >start the flush timer. `<flush_seconds>` later the result will be sent to
+    >IRONdb. It is possible after this initial flush that some late data arrives
+    >that would normally fit into that same aggregation window. This is where
+    >`<idle_cycles>` comes into play. The relay will retain the aggregation
+    >window until no more matching rows are seen for `<idle_cycles>` cycles. If
+    >matching rows *are* seen the aggregation window is updated with the new
+    >values and the record is re-sent to IRONdb. To control the behavior of
+    >conflicts within the database when this happens please see
+    >the [conflict resolution](configuration.md#conflict_resolution) section of
+    >the IRONdb configuration.
+    
+### send
+
+This config has a single attribute: `durable="true|false"`. If set to "true" it
+will use the `<journal>` settings below to journal every row destined for IRONdb
+nodes. If set to "false", it will bypass the journaling and directly send to
+IRONdb. If set to "false", the relay will do its best to make sure data arrives
+at one of the IRONdb nodes if the primary doesn't respond or is down but there
+is no guarantee of delivery.
+
+### listeners
+
+Libmtev network listener configuration. See the [libmtev listener
+documentation](http://circonus-labs.github.io/libmtev/config/listeners.html).
+
+Each listener below is configured within a `<listener>` node. Additional
+listeners may be configured if desired, or the specific address and/or port may
+be modified to suit your environment.
+
+#### Main listener
+
+```
+<listener address="*" port="8112" backlog="100" type="http_rest_api">
+  <config>
+    <document_root>/opt/circonus/share/snowth-web</document_root>
+  </config>
+</listener>
+```
+
+The main listener serves multiple functions:
+* [HTTP REST API](api.md)
+* [Cluster replication](operations.md#replication) (TCP) and gossip (UDP)
+* [Operations Dashboard](operations.md#operations-dashboard)
+* JSON-formatted node statistics (`http://thisnode:thisport/stats.json`)
+
+##### Main listener address
+
+The IP address on which to listen, or the special `*` to listen on any local IP
+address.
+
+Default: `*`
+
+##### Main listener port
+
+The port number to listen on. For the main listener this will utilize both TCP
+and UDP.
+
+Default: 8112
+
+##### Main listener backlog
+
+The size of the queue of pending connections. This is used as an argument to
+the standard `listen(2)` system call. If a new connection arrives when this
+queue is full, the client may receive an error such as ECONNREFUSED.
+
+Default: 100
+
+##### Main listener type
+
+The type of libmtev listener this is. The main listener is configured to be
+only a REST API listener. This value should not be changed.
+
+Default: http_rest_api
+
+#### Graphite listener
```

Comment:
Are the graphite listeners supposed to match what's in IRONdb?  If so, maybe mention that they should be kept in sync?